eZ/Publish/Core/Persistence/Legacy/Content/Type/MemoryCachingHandler.php

```diff
@@ -169,6 +169,20 @@ public function loadContentTypes($groupId, $status = 0)
         return $this->innerHandler->loadContentTypes($groupId, $status);
     }
 
+    public function loadContentTypeList(array $contentTypeIds): array
+    {
+        $contentTypes = [];
+        foreach ($contentTypeIds as $contentTypeId) {
+            if (isset($this->contentTypes[$contentTypeId])) {
+                $contentTypes[$contentTypeId] = $this->contentTypes[$contentTypeId];
+            } else {
+                $contentTypes[$contentTypeId] = $this->contentTypes[$contentTypeId] = $this->innerHandler->load($contentTypeId);
```

Comment:
should improve this here and in loadGroups as otherwise it means singular load is always done.